src/redux/reducers/productSlice.ts

```diff
@@ -0,0 +1,70 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+
+import { Product } from '../../models/product.model';
+
+type InitialStateType = {
+    products: (Product & { count: number })[];
+};
+
+const initialState: InitialStateType = {
+    products: [],
+};
+
+export const productSlice = createSlice({
+    name: 'products',
+    initialState,
+    reducers: {
+        removeProduct: (state, { payload }: PayloadAction<{ id: number }>) => {
+            const foundProduct = state.products.find(
+                (product) => product.id === payload.id
+            );
+            if (!foundProduct) return;
+            if (foundProduct.count >= 2) {
+                state.products = state.products.map((product) =>
+                    product.id === payload.id
+                        ? {
+                              id: product.id,
+                              name: product.name,
+                              price: product.price,
+                              count: product.count - 1,
+                          }
+                        : product
+                );
+            } else
+                state.products = state.products.filter(
+                    (product) => product.id !== payload.id
+                );
+        },
+
+        addProduct: (state, { payload }: PayloadAction<Product>) => {
+            const isProductInCart = state.products.find(
```

Comment:
tutaj trzeba zawołać endpoint PUT /orders/{{orderId}/add,

chyba że jest to pierwszy produkt w zamówieniu, wtedy trzeba stworzyć zamówienie POST /orders/

Reply:
opcjonalnie możesz zawsze tworzyć Order, kiedy user wejdzie do restauracji (z pustą listą produktów), a potem wołać same add i remove

Reply:
Ok